File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <cmath>

using namespace std;

int solution(string word) {
    int answer = 0;
    unordered_map<char, int> V;
    V['A'] = 1;
    V['E'] = 2;
    V['I'] = 3;
    V['O'] = 4;
    V['U'] = 5;
    while(word.length() != 5)
    {
        word += " ";
    }
    for(int i = 0; i < word.length();i++)
    {
        if(word[i] == ' ') break;
        int temp = 0;
        for(int j = 1; j < word.size()-i; j++)
        {
            temp += pow(5, j);
        }
        answer += V[word[i]] + temp * (V[word[i]]-1);
    }
    return answer;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(string word) {
        int answer = 0;
        Dictionary<char, int> dic = new();
        dic['A'] = 1;
        dic['E'] = 2;
        dic['I'] = 3;
        dic['O'] = 4;
        dic['U'] = 5;
        while(word.Length != 5)
        {
            word = $"{word} ";
        }
        for(int i = 0; i < word.Length; i++)
        {
            if(word[i] == ' ') break;
            int temp = 0;
            for(int j = 1; j < dic.Count-i; j++)
            {
                temp += (int)Math.Pow(5, j);
            }
            answer += dic[word[i]] + temp * (dic[word[i]]-1);
        }
        return answer;
    }
}
